### ISSUE TITLE
Keyboard layout widget is practically unusable 

### ISSUE THREAD

**Issue by rosenpin** [Jean-Tinland/simple-bar#199]
The current query delay in the keyboard layout widget is too long.  
If I change the layout, the widget will not reflect the change for a long time, a better way to do this would be to listen to the system event of keyboard layout change and only update then.  

In my fork I just changed it to 1 second refresh rate, which is very inefficient

**Comment by rosenpin** [comment]
This could help  
https://stackoverflow.com/a/15169594

**Comment by Jean-Tinland** [comment]
Yes this widget could really be improved, I'll look into it, thanks! :)

**Comment by jaideeprana** [comment]
Eagerly waiting for this feature/enhancement.

**Comment by lclrc** [comment]
> This could help
> [stackoverflow.com/a/15169594](https://stackoverflow.com/a/15169594)

Looks great!

**Comment by Jean-Tinland** [comment]
I'm closing this issue in favor of this one (#205). A system that would allow to trigger refresh on a specific widget like the keyboard one would resolve the issue.
